src/main/java/com/amazon/opendistroforelasticsearch/sql/exception/SqlFeatureNotImplementedException.java

```diff
@@ -0,0 +1,19 @@
+package com.amazon.opendistroforelasticsearch.sql.exception;
+
+/**
+ * Intended for cases when we knowingly omitted some case, letting users know that we didn't implemented feature, but
+ * it may be implemented in future.
+ */
+public class SqlFeatureNotImplementedException extends RuntimeException {
+    private static final long serialVersionUID = 1;
+
+    public SqlFeatureNotImplementedException() { super();}
```

Comment:
I still think we need a default message giving some context about what went wrong, otherwise we will have to enforce always catching this and converting to a more user-friendly response/exception. I would even prefer not to provide the default constructor if you feel there's not a single good message.

Reply:
It's just typical pattern, look at the implementation of 

```
 */
public
class IllegalArgumentException extends RuntimeException {
    /**
     * Constructs an <code>IllegalArgumentException</code> with no
     * detail message.
     */
    public IllegalArgumentException() {
        super();
    }
```
for example

Reply:
`IllegalArgumentException` is such a common type of exception, and since it's used in different layers, so the type without any extra info is enough to do things like

```
... } catch (IllegalArgumentException e) {
...
} catch (OtherException e) {
...
}
```
What I am saying, at least from the description of the class, your exception is user-facing, so it doesn't make much sense having it without any messaging unless someone will always do
```
catch (SqlFeatureNotImplementedException e) {
   return buildResponse(e); // made up function for illustration
}
```

Finally, maybe it's better [not to add it if we're not using it](https://www.martinfowler.com/bliki/Yagni.html).

Up to you if you want to keep it as is, just seems a bit off to me.

Reply:
Fixed